src/main/java/org/jenkinsci/pipeline_steps_doc_generator/ProcessAsciiDoc.java

```diff
@@ -0,0 +1,93 @@
+package org.jenkinsci.pipeline_steps_doc_generator;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileReader;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.List;
+
+import org.apache.commons.io.FilenameUtils;
+
+public class ProcessAsciiDoc {
+
+    public void generateHeader(StringBuilder toWrite, String param) {
+        toWrite.append("---\nlayout: pipelinesteps\ntitle: \"").append(param)
+                .append("\"\n---\n")
+                .append("== " + param + "\n")
+                .append("\n++++\n");
+    }
+
+    /*
+     * Takes in param name, and separates its documentation on to a new file.
+     */
+    public String separateParam(String param, String allAscii, File child)
+            throws IOException, RuntimeException {
+        BufferedReader br = new BufferedReader(new FileReader(child));
```

Comment:
I know you've declared that this throws an `IOException`, but you should catch this here since you're working with the file, and you can close the FileReaders in a finally statement.